File: CS50's X/WEEK 2/readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <stdio.h>
#include <string.h>

int main(void)
{
    string text = get_string("Text: ");

    int length = strlen(text);
    float words = 1;
    float letters = 0;
    float sentences = 0;

    for (int i = 0; i < length; i++)
    {
        if (isalpha(text[i]))
        {
            letters++;
        }
        else if (isblank(text[i]))
        {
            words++;
        }
        else if (text[i] == '?' || text[i] == '!' || text[i] == '.')
        {
            sentences++;
        }
    }

    int index = 0.0588 * (letters / words) * 100 - 0.296 * (sentences / words) * 100 - 15.8 + 0.5;

    if (index < 1)
    {
        printf("Before Grade 1\n");
    }

    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }

    else
    {
        printf("Grade %i\n", index);
    }
}

File: CS50's X/WEEK 6/sentimental-readability/readability.py
from cs50 import get_string


def main():
    text = get_string("Text: ")

    words = 1
    letters = 0
    sentences = 0

    for c in text:
        if c.isalpha():
            letters += 1
        elif c.isspace():
            words += 1
        elif c in ['?', '!', '.']:
            sentences += 1

    index = round(0.0588 * (letters / words) * 100 - 0.296 * (sentences / words) * 100 - 15.8)

    if index < 1:
        print("Before Grade 1")
    elif index >= 16:
        print("Grade 16+")
    else:
        print(f"Grade {index}")


main()
